contracts/AllDay.cdc

```diff
@@ -558,6 +569,269 @@ access(all) contract AllDay: NonFungibleToken {
         AllDay.account.storage.save(setPlayTierMap, to: /storage/AllDayAdminSetPlayTierMap)
     }
 
+    //------------------------------------------------------------
+    // Badges
+    //------------------------------------------------------------  
+
+    access(all) struct Badge{
+        access(all) var slug: String
+        access(all) var title: String
+        access(all) var description: String
+        access(all) var visible: Bool
+        access(all) var slugV2: String
+        access(all) var metadata: {String: String}
+
+        init(slug: String, title: String, description: String, visible: Bool, slugV2: String){
+            self.slug = slug
+            self.title = title
+            self.description = description
+            self.visible = visible
+            self.slugV2 = slugV2
+            self.metadata = {}
+        }
+
+        access(all) fun update(title: String?, description: String?, visible: Bool?, slugV2: String?, metadata: {String: String}?){
+            if title != nil{
+                self.title = title!
+            }
+            if description != nil{
+                self.description = description!
+            }
+            if visible != nil{
+                self.visible = visible!
+            }
+            if slugV2 != nil{
+                self.slugV2 = slugV2!
+            }
+            if metadata != nil{
+                self.metadata = metadata!
+            }
+        }
+    }
+
+    access(all) entitlement ManageBadges
+
+    access(all) resource Badges {
+        access(all) var slugToBadge: {String: Badge}
+        access(all) var playIdToBadgeSlugs: {UInt64: {String: {String: String}}}
+        access(all) var editionIdToBadgeSlugs: {UInt64: {String: {String: String}}}
+        access(all) var momentIdToBadgeSlugs: {UInt64: {String: {String: String}}}
+        
+
+        // Badges initializer
+        //
+        init() {
+            self.slugToBadge = {}
+            self.playIdToBadgeSlugs = {}
+            self.editionIdToBadgeSlugs = {}
+            self.momentIdToBadgeSlugs = {}
+        }
+
+        access(all) view fun getBadge(_ slug: String): Badge?{
+            return self.slugToBadge[slug]
+        }
+
+        access(all) fun getPlayBadges(_ playID: UInt64): [Badge]?{
```

Comment:
Nit: use generic methods to get/add/remove entity badges (e.g., `EntityType` slice/enum + helper method with switch on type to return the right map)